### DISCUSSION TITLE
mask nc using shapefile and calculate average and anomaly

### DISCUSSION THREAD

**Discussion by shuai-zhou** [pydata/xarray#6072] [Q&A] [answered]
Dear all,

New user to xarray here. This is a cross-posting that can also be found [here](https://stackoverflow.com/questions/70241007/mask-netcdf-using-shapefile-and-calculate-average-and-anomaly-for-all-polygons-w).

Suppose I have the monthly temperature data [download temperature file](https://drive.google.com/file/d/1o-VYs4WsySestmKYjIZXdo2uCAmuPBad/view?usp=sharing) and the state-level US shapefile [download shapefile here](https://drive.google.com/file/d/1G2nVUGWH_ugFbs4BXBzo-oqSkgYP6YCC/view?usp=sharing). How can I get the county-level average temperature for each state like the following table (of course, no need to sort state by name):

|      state      |     avg_temp    | 
|---------------------|------------------|
|          AL         |         xx       |
|          AR         |         xx       |
|          ...        |         ...      |
|          WY         |         xx       |

Also, how can I get the temperature anomaly of the year 2019 relative to baseline years from 2000 to 2010 for each state (like the data format above)? I have been trying with `rioxarray`, `regionmask`, and function proposed by @shoyer [here](https://github.com/pydata/xarray/issues/501#issuecomment-126461466), but was still unclear about the workflow of these tasks. Any help would be much appreciated!

**Comment by mathause** [comment, marked as answer]
Thanks for your question - I think this is only marginally related to xarray so I'll answer on [stackoverflow](https://stackoverflow.com/questions/70241007/mask-netcdf-using-shapefile-and-calculate-average-and-anomaly-for-all-polygons-w).
